modules/karpenter/main.tf

```diff
@@ -259,6 +262,9 @@ locals {
       event_pattern = {
         source      = ["aws.ec2"]
         detail-type = ["EC2 Instance State-change Notification"]
+        detail = {
```

Comment:
This is not what Karpenter has and we follow their guidance https://github.com/aws/karpenter/blob/7bd5857e4df31855b975b3c6383b142d0cd29321/website/content/en/v0.27.0/getting-started/getting-started-with-eksctl/cloudformation.yaml#L133-L143